modules/search/class.jetpack-search-widget-filters.php

```diff
@@ -140,10 +149,94 @@ function widget( $args, $instance ) {
 			do_action( 'jetpack_search_render_filters_widget_title', $title, $args['before_title'], $args['after_title'] );
 		}
 
+		// we need to dynamically inject the sort field into the search box when the search box is enabled, and display
+		// it separately when it's not.
 		if ( ! empty( $instance['search_box_enabled'] ) ) {
 			$this->render_widget_search_form( $instance );
 		}
 
+		$default_sort = isset( $instance['sort'] ) ? $instance['sort'] : self::DEFAULT_SORT;
+		list( $orderby, $order ) = $this->sorting_to_wp_query_param( $default_sort );
+		$current_sort = "{$orderby}|{$order}";
+
+		if ( ! empty( $instance['search_box_enabled'] ) && ! empty( $instance['user_sort_enabled'] ) ) {
+			?>
+			<label class="jetpack-search-sort-wrapper">
+				<?php esc_html_e('Sort by', 'jetpack'); ?>
+				<select name="<?php echo esc_attr( $this->get_field_name( 'sort' ) ); ?>" class="jetpack-search-sort">
+					<?php foreach( $this->get_sort_types() as $sort => $label ) { ?>
+						<option value="<?php echo $sort; ?>" <?php selected( $current_sort, $sort ); ?>>
+							<?php echo $label; ?>
+						</option>
+					<?php } ?>
+				</select>
+			</label> <?php
+		}
+
+		/*
+		 * this JS is a bit complicated, but here's what it's trying to do:
+		 * - find or create a search form
+		 * - find or create the orderby/order fields with default values
+		 * - detect changes to the sort field, if it exists, and use it to set the order field values
+		 */
+		?>
+		<script type="text/javascript">
+			jQuery( document ).ready( function( $ ) {
+				var actionUrl      = <?php echo json_encode( home_url( '/' ) ); ?>,
+					orderByDefault = <?php echo json_encode( $orderby ); ?>,
+					orderDefault   = <?php echo json_encode( $order ); ?>,
+					widgetId       = <?php echo json_encode( $this->id ); ?>,
+					currentSearch  = <?php echo json_encode( isset( $_GET['s'] ) ? $_GET['s'] : '' ); ?>
+
+				var container = $('#' + widgetId);
+				var form = container.find('.jetpack-search-form form');
+				if ( form.length === 0 ) {
+					form = $('<form></form>')
+						.prop({
+							'action': actionUrl,
+							'role': 'search',
+							'method': 'get',
+							'class': 'search-form'
+						});
+					container.append(form);
+				}
+
+				// create hidden fields if necessary
```

Comment:
As a note, the bits of code that add the inputs input don't set the value. If we wanted that to work, we could do something like:

```js
// create hidden fields if necessary
var orderBy = form.find( 'input[name=orderby]');
if ( orderBy.length === 0 ) {
	orderBy = $('<input>')
		.prop( {
			name: 'orderby',
			type: 'hidden',
			value: orderByDefault
		} );
	form.append( orderBy );
} else {
	orderBy.val( orderByDefault );
}

var order = form.find( 'input[name=order]');
if ( order.length === 0 ) {
	order = $('<input>')
		.prop( {
			name: 'order',
			type: 'hidden',
			value: orderDefault
		} );
	form.append( order );
} else {
	order.val( orderDefault );
}
```

That being said, I'm not sure we need the logic for adding the inputs since we echo the inputs when we build the form.

Reply:
Actually, we set the value further down the function - see the lines:

```js
orderBy.val(orderByDefault);
order.val(orderDefault);
```

Reply:
You're 100% right. I misunderstood and thought that bit of JS was run when the select changed.